tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_floor.h

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+#include "sfpi.h"
+#include "noc_nonblocking_api.h"
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+template <bool APPROXIMATION_MODE, int ITERATIONS = 8>
+inline void calculate_floor()
+{
+    for (int d = 0; d < ITERATIONS; d++)
+    {
+        vFloat result = dst_reg[0];
+        vFloat v = result;
+        vInt tmp = float_to_int16(result);
+        result= int32_to_float(tmp);
+        v_if (result > v){
+            result = result - 1;
+        }
+        v_endif;
+        v_if (v < -32768 || v > 32767){
+            result = v;
```

Comment:
For input ranges v < -32768 || v > 32767, the number itself does not contains any decimal values, so we directly assigned it as output, We have tested for all ranges and it works fine, there is no any range limitations here, hence `float_to_int16(result)` conversion is not required for that range.

Reply:
<img width="487" alt="Screenshot 2024-06-03 at 11 48 47 AM" src="https://github.com/tenstorrent/tt-metal/assets/138196495/0c6fef71-bee8-4126-b500-9c0d3d8148ee">

Reply:
For the case above, its safer to use:
```
      vInt tmp = float_to_int16(result);
       result= int32_to_float(tmp);
```

for all ranges, because I am unsure if you tested for something like -10^6 to 10^6, or much larger ranges. For example this value: `2.4882594e+29` has fractions, and is represented by this hex: 0x7049, which fits for bfloat16. But that value is extremely large and not needed for most use cases, but it shows that there are values outside this range v < -32768 || v > 32767 that can possibly have fractions, but they are not sweeped over.

Also please add in the op documentation that this only works with bfloat16

Reply:
I apologize, I made a mistake here, the first cast is to int16 so it will always saturate in the 32kb range and never reach the value I have above. So the implementation above works but it needs to be documented that it is only accurate in your tested range with Pytorch. To cover a better range we would need `float_to_int32` which is not currently supported.